path.c

```diff
@@ -1250,6 +1250,45 @@ char *strip_path_suffix(const char *path, const char *suffix)
 	return xstrndup(path, chomp_trailing_dir_sep(path, path_len));
 }
 
+int is_mount_point_via_stat(struct strbuf *path)
+{
+	size_t len = path->len;
+	unsigned int current_dev;
+	struct stat st;
+
+	if (!strcmp("/", path->buf))
+		return 1;
+
+	strbuf_addstr(path, "/.");
+	if (lstat(path->buf, &st)) {
+		/*
+		 * If we cannot access the current directory, we cannot say
```

Comment:
If we cannot access the current directory, and return false here, will we then proceed to clean files in this directory even though it might actually be a mount point? If so, maybe we should act on the side of caution and check `is_definitely_not_a_mount_point` instead.

Reply:
This is for Linux mount points, right? Let's ignore this is a corner case.